compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -148,6 +148,7 @@ object Types {
     /** Does this type denote a stable reference (i.e. singleton type)? */
     final def isStable(implicit ctx: Context): Boolean = stripTypeVar match {
       case tp: TermRef => tp.termSymbol.isStable && tp.prefix.isStable || tp.info.isStable
+      case tp: AppliedTermRef => tp.fn.isStable && tp.args.forall(_.isStable)
```

Comment:
Aren't all AppliedTermRefs stable by definition?

Reply:
I'd think they can't be stable if the arguments aren't stable; OTOH, I suspect `AppliedTermRef` with unstable arguments shouldn't be just unstable but illegal.

Reply:
That's what I meant.

Reply:
In principle `f(unstableThing)` is still meaningful, if you think of it as the image of `f` on set `unstableThing`. E.g., we could still type `sign(impure()): 1.type` where `def impure(): PositiveInt`. That being said, I think it's fine to just make `AppliedTermRef`s stable by construction for now, and I'll come back to this in my branch if I find a compelling use-case.